libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m

```diff
@@ -932,39 +931,43 @@ - (void)resetWebUserAgent
 
 - (void)startRefreshFlowConnectionTimer
 {
-    self.refreshTimerThread = [NSThread currentThread];
+    if (![NSThread isMainThread]) {
+        __weak typeof(self) weakSelf = self;
```

Comment:
`dispatch_async()` will not capture `self`, and in this case it's possible to lose the reference to `self` as a result.  I'd remove the casting here.